float64 angle
---
bool ret
